src/cmd/run.js

```diff
@@ -44,16 +49,25 @@ export function defaultWatcherCreator(
   {
     addonId, client, sourceDir, artifactsDir,
     onSourceChange = defaultSourceWatcher,
+    messenger = notifier,
   }: WatcherCreatorParams
  ): Watchpack {
   return onSourceChange({
     sourceDir,
     artifactsDir,
     onChange: () => {
       log.debug(`Reloading add-on ID ${addonId}`);
+      messenger.notify({
+        title: 'Started reloading',
+        message: `Reloading add-on ID ${addonId}`,
+      });
       return client.reloadAddon(addonId)
         .catch((error) => {
```

Comment:
It would be probably better to notify the user that the addon has been reloaded successfully (the success scenario) instead of sending a notification before starting the reload, so that we alway show 1 notification per addon reload (with the current implementation we are going to show 1 notification when the addon reload successfully and 2 notifications on reloading errors).

We can achieve it by running the success notify in a `then` callback before the above `catch`, eg.

```
return client.reloadAddon(addonId)
  .then(() => {
     // notify addon successfully reloaded! 
     // this is only called if the promise returned by reloadAddon has not been rejected
  })
  .catch((error) => {
     // notify addon reloading error! 
     // this is called when the promise returned by reloadAddon has been rejected, or if the above callback
     // raises an exception.
  });
```